nextjs/src/app/(auth)/account/newpassword/new-password-form.tsx

```diff
@@ -0,0 +1,123 @@
+"use client";
+
+import { OperationVariables, gql, useMutation } from "@apollo/client";
+import { useForm } from "react-hook-form";
+import { useState } from "react"
+import { zodResolver } from "@hookform/resolvers/zod";
+import { z } from "zod";
+import { useSearchParams } from 'next/navigation';
+import { Button } from "@/components/ui/button";
+import {
+    Form,
+    FormControl,
+    FormField,
+    FormItem,
+    FormLabel,
+    FormMessage,
+} from "@/components/ui/form";
+import { Input } from "@/components/ui/input";
+
+const PERFORM_PASSWORD_RESET_MUTATION = gql`
+  mutation PerformPasswordReset($token: String!, $password1: String!, $password2: String!) {
+    performPasswordReset(token: $token, newPassword1: $password1, newPassword2: $password2) {
+      errors
+      success
+    }
+  }
+`;
+
+export default function NewPasswordForm() {
+    const searchParams = useSearchParams();
+    const token = searchParams?.get('token');
+
+    // TODO make this schema more secure
```

Comment:
question: what's insecure about it?

Reply:
I mean the password schema. Should we impose conditions on password e.g. number of letters and numbers (it's already got to be 8 characters long)

Reply:
Ah right – I think there is already a schema for this in the register form:

```javascript
 password1: z
            .string()
            .min(8)
            .regex(
              new RegExp(
                /^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).{8,}$/,
              ),
              {
                message:
                  "Your password must include at least one uppercase letter, one lowercase letter, one number and one special character, and be at least 8 characters long.",
              },
            ),
```

Reply:
cool, so we can remove the validation here entirely?